pipeline/readme.md

```diff
@@ -0,0 +1,137 @@
+**requirements.txt**: it defines what python libraries will be installed in the docker container.  
+**sample_pipeline.py**: the example pipeline class, where it loads two models, and makes prediction using all models.  
+**utils.py**: it defines all the utility functions that will be used in the **pipeline_class.py**.  
+
+
+## Pipeline API
+The gadget pipeline is the class to load the configurations (confidence levels, trained AI models paths, class map, etc.), load and warm up the models, make predictions, and clean up the models. 
+To complete these taks, below are the required methods to be implemented:
+1. **def \_\_init\_\_(self, `**kwargs`) -> None:**  
+    This is the function to load and initialize the pipeline class configurations, where `kwargs` are keyword arguments and it contains the key-value pairs that are defined in the **pipeline_def.json**.
+2. **def load(self) -> None:**  
+    This function loads the models.
+3. **def warm_up(self) -> None:**  
+    This function run the models the first time.
+4. **def predict(self, `input_image`: str, `configs`: dict, `results_storage_path`: str) -> dict:**  
+    This function loads the `input_image` and `configs`, make predictions, add annotations to the `input_image`, save the annotated image to the `results_storage_path`. This function must return a dictionary. See the details in section [Pipeline Result Dictionary](#pipeline-result-dictionary).  
+    Besides, It would be helpful for developers to define a unit test in the main function. Refer to the main function in the **sample_pipeline.py**. 
+5. **def clean_up(self):**  
+    This function deletes the models from memory.
+
+## Pipeline Result Dictionary
+The pipeline predict method will return a dictionary that follows this pattern.  
+```python
+{
+    # Required 
+    "file_path": string,
+    "automation_keys": list,
+    "factory_keys": list,
+    "tags":list,
+    "should_archive":bool,
+
+    # Custom
+    "key-1": VALUE,
+    "key-2": VALUE,
+    "key-3": VALUE,
+}
+```
+The key-value pairs in the dictionary must contain a literal or a list. No other object is allowed. The key-value pairs are grouped into two categories: **Required** and **Custom**.   
+
+For the **Required** key-value pairs, the value of `file_path` is a string of the annotated output image. The value of `automation_keys` is a list of strings that are a subset of **Custom** keys that will be sent to the automation service. They should be information that is specifically needed by the automation service. The value of `factory_keys` is a list of strings that are a subset of **Custom** keys that will be consumed by goFactory. They should be values that are actually useful in goFactory and will not overly inflate the size of the database. The value of `tags` should be a list of strings. They could be the predicted class names. The value of `should_archive` is a boolean indicating whether archive this input or not.  
```

Comment:
**the value of `file_path` is a string of the annotated output image**
Add "path" to end of this sentence. 

**is a boolean indicating whether archive this input or not**
Should be whether to archive